printing_auto_base/models/printing_auto.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2022 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
+# Copyright 2022 Michael Tietz (MT Software) <mtietz@mt-software.de>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.osv import expression
+from odoo.tools.safe_eval import safe_eval
+
+
+class PrintingAuto(models.Model):
+    """Configure which document to print automatically. This model must be
+    linked with a many2many relation from the another model from which you want
+    to print a document"""
+
+    _name = "printing.auto"
+    _description = "Printing Auto"
+
+    name = fields.Char(string="Name", required=True)
+
+    data_source = fields.Selection(
+        [
+            ("report", "Report"),
+            ("attachment", "Attachment"),
+        ],
+        string="Data source",
+        default="report",
+        required=True,
+        help=(
+            "Choose to print the result of an odoo report or a pre-existing "
+            "attachment (useful for labels received from carriers that are "
+            "recorded on the picking as an attachment)"
+        ),
+    )
+    report_id = fields.Many2one("ir.actions.report")
+    attachment_domain = fields.Char("Attachment domain", default="[]")
+
+    condition = fields.Char(
+        "Condition",
+        default="[]",
+        help="Give a domain that must be valid for printing this",
+    )
+    record_change = fields.Char(
+        "Record change", help="Select on which document the report must be executed"
```

Comment:
This help msg must be improved, is not clear what it's used for and how it should be valued.